src/databricks/labs/ucx/azure/credentials.py

```diff
@@ -136,7 +161,12 @@ def validate(self, permission_mapping: StoragePermissionMapping) -> StorageCrede
                 continue
             if result.result == ValidationResultResult.FAIL:
                 failures.append(f"{result.operation.value} validation failed with message: {result.message}")
-        return StorageCredentialValidationResult.from_validation(permission_mapping, None if not failures else failures)
+
+        return StorageCredentialValidationResult.from_storage_credential_info(
+            storage_credential_info,
+            url,
+            None if not failures else failures,
```

Comment:
shouldn't `[]` be enough?

Reply:
I think so too, it was already here, so kept it. I updated it now, I prefer the empty list over the `None`